spec/lib/brexit_checker/question_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+
+describe BrexitChecker::Question do
+  let(:q1) { FactoryBot.build(:brexit_checker_question) }
+  let(:q2) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(a b) }]) }
+  let(:q3) { FactoryBot.build(:brexit_checker_question, criteria: [{ "all_of" => %w(c d) }]) }
+  let(:q4) { FactoryBot.build(:brexit_checker_question, criteria: %w(c)) }
+
+  let(:questions) { [q1, q2, q3, q4] }
+
+  describe '#resolve' do
+    before do
+      allow(BrexitChecker::Question).to receive(:load_all) { questions }
+    end
+
+    let(:subject) {
+      BrexitChecker::Question.resolve(
+        previous_question: previous_question, criteria_keys: criteria_keys
+      )
+    }
+
+    let(:criteria_keys) { [] }
+    let(:previous_question) { nil }
+
+    context 'the page on the form is 0' do
+      let(:previous_question) { 0 }
+      it 'returns question one ' do
+        expect(subject).to eq(q1)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys match question_two' do
+      let(:previous_question) { 1 }
+      let(:criteria_keys) { %w[c d] }
+      it 'returns question three' do
+        expect(subject).to eq(q3)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys match question_three' do
+      let(:previous_question) { 1 }
+      let(:criteria_keys) { %w[c] }
+      it 'returns question four' do
+        expect(subject).to eq(q4)
+      end
+    end
+
+    context 'the page on the form is 1 and the criteria_keys do not match anything' do
+      let(:previous_question) { 1 }
+      let(:criteria_keys) { %w[e] }
+      it 'returns nil' do
```

Comment:
Could we be more explicit here, and in the next test, to indicate that we've reached the end of the questions? I think it's important to be clear about the meaning of `nil` if we're using it to make decisions later.

Reply:
Updated the description of the example - is it a bit clearer?

Reply:
Thanks for updating - it's a bit more clear. I realise I wasn't specific about what was confusing me. The thing I'm struggling with is really the contexts. For example, when I'm reading "previous question id is one and the criteria matches question three", I'm wondering "what does that really mean?", or "how would it appear to the user?". In this case, what we're testing is that we "skip questions with un-met criteria" - this means I'm learning about the overall functionality, and understanding the particular question number is just a detail.

Some other examples I thought of while reading the other contexts:

   - we "return nil when there are no further questions" (vs.)
   - we "return nil when the remaining questions have un-met criteria"